README.md

```diff
@@ -146,18 +146,24 @@ repo](https://github.com/banacorn/docker-agda).
 
 # How To Connect The Paper To This Mechanization
 
-Genearlly, files are named after the theorems that they prove. In
+Generally, files are named after the theorems that they prove. In
 particular, the theorems, figures, and definitions mentioned explicitly in
 Section 4 can be found as follows:
 
 * The contents of Figure 4 can be found in
-  [core.agda:7-107](core.agda#L7-L107).
+  [core.agda:7-107](core.agda#L7-L107). Note that the notation for product
```

Comment:
Maybe say something about how sums and recursive types are only necessary to be able to encode the syntax in the language, but to simply matters (as you described here) we instead postulate that encoding is possible (encoding syntax into a typed lambda calculus is well-understood).

Reply:
Updated in 62d3120!